### PR TITLE
feat(image): add CI compatible Dockerfile

### PR DESCRIPTION
**Pull request by njhale**
### Description
- Adds CI compatible Dockerfile
- Corrects package used for healthcheck in client
- Adds missing fields to expected bundle in configmap test
- Adds `-count=1` to `make test` to prevent test result caching
- Regenerates go.sum

### CODE REVIEW

**Comment by njhale** [comment]
Seeing some issues with protobuf.

**Comment by njhale** [comment]
Seeing some issues with protobuf.

/hold

File: Dockerfile [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
+
+WORKDIR /go/src/github.com/operator-framework/operator-registry
 
 COPY vendor vendor
 COPY cmd cmd
 COPY pkg pkg
 COPY Makefile Makefile
-COPY go.mod go.mod
 RUN make static
+RUN echo $(ls -la bin)
```
Comment (ecordell) [inline Dockerfile]: remove?

File: Makefile [hunk 2]
```diff
@@ -1,23 +1,28 @@
 CMDS  := $(addprefix bin/, $(shell go list ./cmd/... | xargs -I{} basename {}))
+MOD_FLAGS := $(shell (go version | grep -q 1.11) && echo -mod=vendor)
 
 .PHONY: build test vendor clean
 
 all: clean test build
 
 $(CMDS):
-	go build -mod=vendor -tags json1 $(extra_flags) -o $@ ./cmd/$(shell basename $@)
+	CGO_ENABLE=0 go build $(MOD_FLAGS) -tags json1 $(extra_flags) -o $@ ./cmd/$(shell basename $@)
```
Comment (ecordell) [inline Makefile]: sqlite bindings require CGO: https://github.com/mattn/go-sqlite3#installation

File: pkg/client/client.go [hunk 3]
```diff
@@ -15,31 +18,31 @@ type Interface interface {
 
 type Client struct {
 	client api.RegistryClient
-	health api.HealthClient
-	conn *grpc.ClientConn
+	health grpc_health_v1.HealthClient
```
Comment (ecordell) [inline pkg/client/client.go]: oops!

**Pull request merged**